### ISSUE TITLE
New feature suggestions

### ISSUE THREAD

**Issue by lanzhixi** [Tauffer-Consulting/domino#94]
This is an impressive project. During my usage, I've been looking to encapsulate a workflow into a single piece. This would allow others to directly utilize this piece to construct more advanced workflows. Is this conceptually possible? Would you contemplate adding such a feature？

**Comment by luiztauffer** [comment]
thanks for the suggestion @lanzhixi !
Indeed, this would be an awesome feature, we're thinking about ways to incorporate that in the future. Conceptually it is possible

**Comment by lanzhixi** [comment]
Thank you for your affirmation. Do you have any ideas on how to implement this feature? I'd like to give it a try, although I can't guarantee success. Thank you!

**Comment by lanzhixi** [comment]
thank you!

**Comment by lanzhixi** [comment]
Hello @luiztauffer ,I have learned some front-end and back-end knowledge during this period, as well as related content about Airflow. I found that task groups (https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#taskgroups) seem to be able to encapsulate processes. 

I roughly understand the process of front-end and back-end interaction. When the front-end clicks save, it creates a new workflow through the request /workspaces/{workspace_id}/workflows. The create_workflow function in workflow_router.py on the backend is called, thus generating a new workflow. In the process of creating a workflow on the backend, the _create_dag_code_from_raw_json function in workflow_service.py is called to convert the json in the body into dag code. I am thinking whether it's possible to add support for task groups in this function, so as to support the function of encapsulating processes. Is my idea feasible?

Although I can now roughly understand the process, there is a big gap from actual hands-on, so I would like to ask you to help design the data format for transmission when the front-end and back-end interact, as well as the backend part. I can try the front-end part, thank you!!!
